static/app/stores/Commiters/CommittersContext.tsx

```diff
@@ -0,0 +1,279 @@
+import {createContext, useContext, useEffect, useReducer} from 'react';
+
+import {Client} from 'sentry/api';
+import {
+  AvatarProject,
+  Committer,
+  Event,
+  Group,
+  Organization,
+  Project,
+} from 'sentry/types';
+import getDisplayName from 'sentry/utils/getDisplayName';
+import useApi from 'sentry/utils/useApi';
+
+function exhaustive(x?: never) {
+  throw new Error(
+    `Unhandled ${JSON.stringify(x)} switch case action in committersReducer`
+  );
+}
+
+function makeCommitterStoreKey({
+  organizationSlug,
+  projectSlug,
+  eventId,
+}: {
+  eventId: string;
+  organizationSlug: string;
+  projectSlug: string;
+}): string {
+  return `${organizationSlug} ${projectSlug} ${eventId}`;
+}
+
+export function fetchCommitters(
+  api: Client,
+  params: {eventId: string; organizationSlug: string; projectSlug: string}
+): Promise<{committers: Committer[]}> {
+  return api.requestPromise(
+    `/projects/${params.organizationSlug}/${params.projectSlug}/events/${params.eventId}/committers/`,
+    {
+      method: 'GET',
+    }
+  );
+}
+
+type ResetAction = {
+  type: 'reset';
+};
+
+type SetCommittersError = {
+  payload: {
+    eventId: string;
+    organizationSlug: string;
+    projectSlug: string;
+  };
+  type: 'set error';
+};
+
+type StartLoadingCommitters = {
+  payload: {
+    eventId: string;
+    organizationSlug: string;
+    projectSlug: string;
+  };
+  type: 'start loading';
+};
+
+type AddCommitters = {
+  payload: {
+    committers: Committer[];
+    eventId: string;
+    organizationSlug: string;
+    projectSlug: string;
+  };
+  type: 'add committers';
+};
+
+export type CommittersState = Record<
```

Comment:
```suggestion
 type CommittersState = Record<
```

Reply:
Not here, but in the later PR's we do yes